Client/Frontend/Browser/URLBarView.swift

```diff
@@ -284,17 +305,7 @@ class URLBarView: UIView {
     // that can show in either mode.
     func setShowToolbar(shouldShow: Bool) {
         toolbarIsShowing = shouldShow
-        setNeedsUpdateConstraints()
-    }
-
-    func updateURLBarText(text: String) {
-        delegate?.urlBarDidBeginEditing(self)
-
-        locationView.text = text
-
-        updateLayoutForEditing(editing: true)
-
-        delegate?.urlBar(self, didEnterText: text)
+        updateViewsForOverlayModeAndToolbar()
```

Comment:
Maybe i should make it even longer...`updateViewsForOverlayModeAndToolbarChanges` :) I was trying to clearly name the factors that affect it, so it's more obvious when to call it. Vague methods name make the code a lot more confusing.